src/drivers/ports/sam/i2c_port.i

```diff
@@ -0,0 +1,544 @@
+#define SAM_TWI_SR_ENDTX        (1 << 13) /* Master: End of TX Buffer - Transmit Counter Register has reached 0 since the last write in TWI_TCR or TWI_TNCR */
+#define SAM_TWI_SR_RXBUFF       (1 << 14) /* Master: RX Buffer Full:  Both TWI_RCR and TWI_RNCR have a value of 0. */
+#define SAM_TWI_SR_TXBUFF       (1 << 15) /* Master: TX Buffer Full:  Both TWI_TCR and TWI_TNCR have a value of 0. */
+
+
+/* TWI_IER - Interrupt Enable Register */
+#define SAM_TWI_IER_TXCOMP      (1)
+#define SAM_TWI_IER_RXRDY       (1 << 1)
+#define SAM_TWI_IER_TXRDY       (1 << 2)
+
+#define SAM_TWI_IER_SVACC       (1 << 4)
+#define SAM_TWI_IER_GACC        (1 << 5)
+#define SAM_TWI_IER_OVRE        (1 << 6)
+
+#define SAM_TWI_IER_NACK        (1 << 8)
+#define SAM_TWI_IER_ARBLST      (1 << 9)
+#define SAM_TWI_IER_SCLWS       (1 << 10)
+#define SAM_TWI_IER_EOSACC      (1 << 11) 
+#define SAM_TWI_IER_ENDRX       (1 << 12)
+#define SAM_TWI_IER_ENDTX       (1 << 13)
+#define SAM_TWI_IER_RXBUFF      (1 << 14)
+#define SAM_TWI_IER_TXBUFF      (1 << 15)
+
+
+/* TWI_IDR - Interrupt Disable Register */
+#define SAM_TWI_IDR_TXCOMP      (1)
+#define SAM_TWI_IDR_RXRDY       (1 << 1)
+#define SAM_TWI_IDR_TXRDY       (1 << 2)
+
+#define SAM_TWI_IDR_SVACC       (1 << 4)
+#define SAM_TWI_IDR_GACC        (1 << 5)
+#define SAM_TWI_IDR_OVRE        (1 << 6)
+
+#define SAM_TWI_IDR_NACK        (1 << 8)
+#define SAM_TWI_IDR_ARBLST      (1 << 9)
+#define SAM_TWI_IDR_SCLWS       (1 << 10)
+#define SAM_TWI_IDR_EOSACC      (1 << 11) 
+#define SAM_TWI_IDR_ENDRX       (1 << 12)
+#define SAM_TWI_IDR_ENDTX       (1 << 13)
+#define SAM_TWI_IDR_RXBUFF      (1 << 14)
+#define SAM_TWI_IDR_TXBUFF      (1 << 15)
+
+/* TWI_IMR - Interrupt Mask Register */
+#define SAM_TWI_IMR_TXCOMP      (1)
+#define SAM_TWI_IMR_RXRDY       (1 << 1)
+#define SAM_TWI_IMR_TXRDY       (1 << 2)
+
+#define SAM_TWI_IMR_SVACC       (1 << 4)
+#define SAM_TWI_IMR_GACC        (1 << 5)
+#define SAM_TWI_IMR_OVRE        (1 << 6)
+
+#define SAM_TWI_IMR_NACK        (1 << 8)
+#define SAM_TWI_IMR_ARBLST      (1 << 9)
+#define SAM_TWI_IMR_SCLWS       (1 << 10)
+#define SAM_TWI_IMR_EOSACC      (1 << 11) 
+#define SAM_TWI_IMR_ENDRX       (1 << 12)
+#define SAM_TWI_IMR_ENDTX       (1 << 13)
+#define SAM_TWI_IMR_RXBUFF      (1 << 14)
+#define SAM_TWI_IMR_TXBUFF      (1 << 15)
+
+/* TWI_RHR - RXDATA: Master or Slave Receive Holding Data*/
+#define SAM_TWI_RHR_MASK        (0xff)
+
+/* TWI_THR - TXDATA: Master or Slave Transmit Holding Data*/
+#define SAM_TWI_THR_MASK        (0xff)
+
+
+int i2c_port_module_init()
+{
+    return (0);
+}
+
+int i2c_port_init(struct i2c_driver_t *self_p,
+                  struct i2c_device_t *dev_p,
+                  int baudrate,
+                  int address)
+{
+    self_p->dev_p = dev_p;
+    self_p->twbr = baudrate;
+    self_p->address = address; /* Perhaps only used for i2c slave mode? */
```

Comment:
Remove the comment. See function declaration in `i2c.h` for the parameter description.